source/modulo_components/src/Component.cpp

```diff
@@ -4,7 +4,48 @@ using namespace modulo_new_core::communication;
 
 namespace modulo_components {
 
-Component::Component(const rclcpp::NodeOptions& node_options) :
-    ComponentInterface<rclcpp::Node>(node_options, PublisherType::PUBLISHER) {}
+Component::Component(const rclcpp::NodeOptions& node_options, bool start_thread) :
+    ComponentInterface<rclcpp::Node>(node_options, PublisherType::PUBLISHER), started_(false) {
+  this->add_predicate("in_error_state", false);
+  this->add_predicate("is_finished", false);
 
-}
\ No newline at end of file
+  if (start_thread) {
+    this->start_thread();
+  }
+}
+
+void Component::start_thread() {
+  if (this->started_) {
+    RCLCPP_ERROR_STREAM_THROTTLE(this->get_logger(), *this->get_clock(), 1000,
+                                 "Run thread for component '" << this->get_name() << "has already been started");
+    return;
+  }
+  this->started_ = true;
+  this->run_thread_ = std::thread([this]() { this->run(); });
+  this->run_thread_.join();
```

Comment:
This will make `start_thread` a blocking function, at which point it is equivalent to calling `run()` directly. We shouldn't block the constructor from finishing if `start_thread` is true.
```suggestion
```